mypy/types.py

```diff
@@ -2123,7 +2121,11 @@ def __init__(
         #   * If it is a non-decorated function, FuncDef is the definition
         #   * If it is a decorated function, enclosing Decorator is the definition
         self.definition = definition
-        self.variables = variables
+        self.variables: tuple[TypeVarLikeType, ...]
+        if variables is None:
+            self.variables = ()
+        else:
+            self.variables = tuple(variables)
```

Comment:
Note, this does crash the `pydantic.v1` extension.
```py
  File "/home/runner/work/ha-core/ha-core/venv/lib/python3.13/site-packages/pydantic/v1/mypy.py", line 868, in add_method
    signature.variables = [tvar_def]
    ^^^^^^^^^^^^^^^^^^^
TypeError: tuple object expected; got list
```

https://github.com/pydantic/pydantic/blob/v2.11.7/pydantic/v1/mypy.py#L866-L868

It can fairly easily be fixed and I'll open a PR for it soon. However, that might not be the only plugin which depends on it being "just" a sequence.

Reply:
If we have evidence that this impacts multiple plugins, we can consider reverting this change.